pkg/transparentproxy/config/config_executables_linux.go

```diff
@@ -39,12 +39,9 @@ func setupSandbox(netns ns.NetNS, needLock bool) error {
 		return errors.Wrap(err, "failed to remount root filesystem as private")
 	}
 
-	if needLock {
+	if c.NeedLock() && c.version.LessThan(IptablesVersionWithLockfileEnv) {
```

Comment:
A random question, when `XTABLES_LOCKFILE` is not available and the container has a readonly filesystem, what will happen? 

We are using this environment variable on `kuma-init` too, I'm thinking if we need to do a similar detection to decide whether we should enable the `readonlyFilesystem` security flag on the container.
https://github.com/kumahq/kuma/blob/dc1daadf0a8ee930bb9d1412a56e10c5a92d4d65/pkg/plugins/runtime/k8s/webhooks/injector/injector.go

Reply:
> A random question, when XTABLES_LOCKFILE is not available and the container has a readonly filesystem, what will happen?

This code runs only in CNI, so as far as I know there shouldn't be issues with read-only filesystems

> Further, is it correct to just specify a new path by using this environment variable on newer iptable versions?

Yes

> Are there other steps steps we need to do on this file, like is there any specified content/format this file should be?

If you are asking about xtables.lock file, then we don't touch the file, we just provide path and kernel is responsible to handle (create, delete etc.) it

> We are using this environment variable on kuma-init too, I'm thinking if we need to do a similar detection to decide whether we should enable the readonlyFilesystem security flag on the container.
https://github.com/kumahq/kuma/blob/dc1daadf0a8ee930bb9d1412a56e10c5a92d4d65/pkg/plugins/runtime/k8s/webhooks/injector/injector.go

For init containers it's different, because it's we who are responsible for installing iptables, so we know what version will be installed: https://github.com/kumahq/kuma/blob/31832e83137324b1b42b504bb20e5871f7fffb29/tools/releases/dockerfiles/kuma-init.Dockerfile#L6

Reply:
Thank you for your detailed answers. They really help resolve my concerns!